src/Data/Json/Extended/Signature/Json.purs

```diff
@@ -0,0 +1,153 @@
+module Data.Json.Extended.Signature.Json where
+
+import Prelude
+
+import Control.Alt ((<|>))
+import Control.Bind ((>=>))
+
+import Data.Argonaut.Core as JS
+import Data.Argonaut.Encode (encodeJson)
+import Data.Argonaut.Decode (class DecodeJson, decodeJson)
+import Data.Argonaut.Combinators ((.?))
+
+import Data.Array as A
+import Data.Either as E
+import Data.HugeNum as HN
+import Data.Int as Int
+import Data.List as L
+import Data.Maybe as M
+import Data.StrMap as SM
+import Data.Traversable as TR
+import Data.Tuple as T
+
+import Data.Json.Extended.Signature.Core (EJsonF(..))
+
+
+encodeJsonEJsonF
+  ∷ ∀ a
+  . (a → JS.Json)
+  → (a → M.Maybe String)
+  → EJsonF a
+  → JS.Json
+encodeJsonEJsonF rec asKey x =
+  case x of
+    Null → JS.jsonNull
+    Boolean b → encodeJson b
+    Integer i → encodeJson i
+    Decimal a → encodeJson $ HN.toNumber a
+    String str → encodeJson str
+    Timestamp str → JS.jsonSingletonObject "$timestamp" $ encodeJson str
+    Time str → JS.jsonSingletonObject "$time" $ encodeJson str
+    Date str → JS.jsonSingletonObject "$date" $ encodeJson str
+    Interval str → JS.jsonSingletonObject "$interval" $ encodeJson str
+    ObjectId str → JS.jsonSingletonObject "$oid" $ encodeJson str
+    OrderedSet xs → JS.jsonSingletonObject "$set" $ encodeJson $ rec <$> xs
+    Array xs → encodeJson $ rec <$> xs
+    Object xs → JS.jsonSingletonObject "$obj" $ encodeJson $ asStrMap xs
```

Comment:
Agreed. I think EJSON <--> JSON codecs should be described somewhere in quasar repo.